handler/src/main/java/io/netty/handler/ssl/ApplicationProtocolNegotiationHandler.java

```diff
@@ -98,6 +99,20 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (!sslHandlerChecked) {
+            sslHandlerChecked = true;
+            if (ctx.pipeline().get(SslHandler.class) == null) {
+                if (bufferedMessages.isEmpty()) {
+                    ctx.fireChannelRead(msg);
+                } else {
+                    // Add to the buffered messages and just remove ourself. This will take care of keep the correct
+                    // ordering.
+                    bufferedMessages.add(msg);
+                }
```

Comment:
Is there a benefit to having the if-else instead of always adding to the bufferedMessages and removing self?

Reply:
not really I guess... lets simplify